From 08822a88f3e5ea05889e0f4d2da3ca40bcf0ad2f Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Tue, 6 Jun 2023 09:42:33 +0200
Subject: [PATCH 2/4] Read the drive handle as a HANDLE

We still incorrectly pass it on as an int after this call.
---
 src/device.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/device.c b/src/device.c
index c0c14e7..deaf7db 100644
--- a/src/device.c
+++ b/src/device.c
@@ -476,6 +476,7 @@ static int libc_open ( dvdcss_t dvdcss, const char *psz_device )
 #if defined( _WIN32 )
 static int win2k_open ( dvdcss_t dvdcss, const char *psz_device )
 {
+    HANDLE h_fd;
     WCHAR psz_dvd[7] = L"\\\\.\\\0:";
     psz_dvd[4] = psz_device[0];
 
@@ -487,24 +488,25 @@ static int win2k_open ( dvdcss_t dvdcss, const char *psz_device )
      * won't send back the right result).
      * (See Microsoft Q241374: Read and Write Access Required for SCSI
      * Pass Through Requests) */
-    dvdcss->i_fd = (int)
+    h_fd =
                 CreateFileW( psz_dvd, GENERIC_READ | GENERIC_WRITE,
                             FILE_SHARE_READ | FILE_SHARE_WRITE,
                             NULL, OPEN_EXISTING,
                             FILE_FLAG_RANDOM_ACCESS, NULL );
 
-    if( (HANDLE) dvdcss->i_fd == INVALID_HANDLE_VALUE )
-        dvdcss->i_fd = (int)
+    if( h_fd == INVALID_HANDLE_VALUE )
+        h_fd =
                     CreateFileW( psz_dvd, GENERIC_READ, FILE_SHARE_READ,
                                 NULL, OPEN_EXISTING,
                                 FILE_FLAG_RANDOM_ACCESS, NULL );
 
-    if( (HANDLE) dvdcss->i_fd == INVALID_HANDLE_VALUE )
+    if( h_fd == INVALID_HANDLE_VALUE )
     {
         print_error( dvdcss, "failed to open device %s", psz_device );
         return -1;
     }
 
+    dvdcss->i_fd = (int) h_fd;
     dvdcss->i_pos = 0;
 
     return 0;
-- 
2.37.3.windows.1

